src/main/java/net/fabricmc/base/launch/FabricClientTweaker.java

```diff
@@ -65,6 +78,56 @@ public void acceptOptions(List<String> args, File gameDir, File assetsDir, Strin
                 this.args.put(arg, args.get(i + 1));
             }
         }
+        
+        createEarlyDisplay();
+    }
+
+    private void createEarlyDisplay() {
+        File gameDir = new File(args.get("--gameDir"));
+        Properties def = new Properties();
+        def.setProperty("fullscreen", "false");
+        def.setProperty("overrideWidth", "0");
+        def.setProperty("overrideHeight", "0");
+        Properties props = new Properties(def);
+        try {
+            props.load(Files.newReader(new File(gameDir, "options.txt"), StandardCharsets.UTF_8));
+        } catch (FileNotFoundException e) {
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        boolean fullscreen = args.containsKey("--fullscreen") || Boolean.parseBoolean(props.getProperty("fullscreen"));
+        int overrideWidth = Integer.parseInt(props.getProperty("overrideWidth"));
+        int overrideHeight = Integer.parseInt(props.getProperty("overrideHeight"));
+        int width = 854;
+        int height = 480;
+        if (overrideWidth > 0) {
+            width = overrideWidth;
+        } else if (args.containsKey("--width")) {
+            width = Math.max(1, Integer.parseInt(args.get("--width")));
+        }
+        if (overrideHeight > 0) {
+            height = overrideHeight;
+        } else if (args.containsKey("--height")) {
+            height = Math.max(1, Integer.parseInt(args.get("--height")));
+        }
+        try {
+            Display.setDisplayMode(new DisplayMode(width, height));
+            Display.setFullscreen(fullscreen);
+            Display.setResizable(true);
+            Display.setTitle("Minecraft 16w33a");
```

Comment:
As mentioned in IRC, there's no getter for the version I can use in the tweaker other than --version, which is equal to the name of the version loaded in the launcher, like "16w33a-fabric-0.0.13.2"; as suggested I'll try splitting on hyphens, but I don't particularly like this solution either.